### PR TITLE
Add @Transactional annotation to jooby-hibernate

### LINKED ISSUE #1920: TransactionalRequest will make all router slower 10x，can it filter by attribute(@transactional)?
TransactionalRequest make all router slower，can it filter by attribute(@transactional)

example:decorator(new TransactionalRequest("transactional"));

### PR DESCRIPTION
**Pull request by imeszaros**
Closes #1920 

Doc page is coming once the API is settled.

### CODE REVIEW

File: modules/jooby-hibernate/src/main/java/io/jooby/hibernate/TransactionalRequest.java [hunk 1]
```diff
@@ -76,28 +79,63 @@ public TransactionalRequest(@Nonnull SessionRequest sessionRequest) {
...
   }
+
+  private boolean isTransactional(Context context) {
+    Object attribute = context.getRoute().attribute(Transactional.ATTRIBUTE);
+
+    if (attribute == null) {
+      return enabledByDefault;
+    }
+
+    if (!(attribute instanceof Boolean)) {
```
Comment (jknack) [inline modules/jooby-hibernate/src/main/java/io/jooby/hibernate/TransactionalRequest.java:134]: Let's invert here these lines:

```
if (attribute instanceof Boolean) {
   return attribute;
}

throw new IllegalArgumentException("Invalid value for route attribute "
          + Transactional.ATTRIBUTE + ": " + attribute);

```

**Comment by imeszaros** [comment]
I've done the changes. I put the `isTransactional` method to the `Context` to avoid duplication. Now that the annotation is in the core i think that's a good place for it. What do you think?

**Comment by imeszaros** [comment]
Do you have an idea where? I was thinking of the `io.jooby.Route` class too because this has to do something with route attributes.

**Comment by jknack** [comment]
yea there: Route. Also, did you try this with MVC? I think mvc saves simple class name in camel case: `transactional`.

**Comment by imeszaros** [comment]
Ok! Yes, but i'll double check it one more time.

**Comment by imeszaros** [comment]
You're right, it does not work with MVC since I moved the annotation to core, because the `jooby-apt` ignores annotations as attributes beginning with `io.jooby.annotations`.

Possible solutions:
- move to another package (then where?)
- whitelist it by name
- create a meta-annotation like `@RouteAttribute` which is then recognized and allowed by `jooby-apt`

**Comment by jknack** [comment]
oh crap! 3 seems more flexible.

**Comment by imeszaros** [comment]
Done. We forgot about ebean.. :) I added support for that too.

**Comment by jknack** [comment]
Awesome!

**Pull request merged**